internal/server/server.go

```diff
@@ -219,40 +241,36 @@ func postQuery(database knowledge.GraphDB) http.HandlerFunc {
 	}
 }
 
-type GraphSnapshotRequestBody struct {
-	Token string `json:"token"`
-}
-
-func isTokenValid(r *http.Request) (bool, string) {
+func isTokenValid(registry importers.Registry, r *http.Request) (bool, string, error) {
 	token, ok := r.URL.Query()["token"]
 
 	if !ok || len(token) != 1 {
-		return false, ""
+		return false, "", fmt.Errorf("Unable to detect token query parameter")
 	}
 
-	sourceToToken := viper.GetStringMap("sources")
+	importerToToken, err := registry.ListImporters(r.Context())
 
-	found := false
-	sourceName := ""
-
-	for sn, t := range sourceToToken {
-		if v, ok := t.(string); ok && v == token[0] {
-			found = true
-			sourceName = sn
-			break
-		}
+	if err != nil {
+		return false, "", err
 	}
 
-	if !found {
-		return false, ""
+	for sn, t := range importerToToken {
+		if t == token[0] {
+			return true, sn, nil
+		}
 	}
 
-	return true, sourceName
+	return false, "", nil
 }
 
-func getGraphRead(graphDB knowledge.GraphDB) http.HandlerFunc {
+func getGraphRead(registry importers.Registry, graphDB knowledge.GraphDB) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ok, source := isTokenValid(r)
+		ok, source, err := isTokenValid(registry, r)
+		if err != nil {
+			replyWithInternalError(w, err)
```

Comment:
err means missing token => should return a 401 (or HTTP 400) instead of a HTTP 500

Reply:
well in that case it's actually an error but I can.

Reply:
err does not means missing token, "!ok" does.